x-pack/platform/plugins/shared/task_manager/server/lib/get_es_and_so_availability.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Observable, BehaviorSubject } from 'rxjs';
+import { Logger, ServiceStatusLevels, CoreStatus, IClusterClient } from '@kbn/core/server';
+import { TASK_MANAGER_INDEX } from '../constants';
+
+export interface GetElasticsearchAndSOAvailabilityOpts {
+  core$: Observable<CoreStatus>;
+  isServerless: boolean;
+  logger: Logger;
+  getClusterClient: () => Promise<IClusterClient>;
+}
+
+export function getElasticsearchAndSOAvailability({
+  core$,
+  isServerless,
+  logger,
+  getClusterClient,
+}: GetElasticsearchAndSOAvailabilityOpts): Observable<boolean> {
+  let isEsHealthy = false;
+  let isEsServiceAvailable = false;
+  let isSoServiceAvailable = false;
+  const result = new BehaviorSubject<boolean>(false);
+  core$.subscribe(({ elasticsearch, savedObjects }) => {
+    isEsServiceAvailable = elasticsearch.level === ServiceStatusLevels.available;
+    isSoServiceAvailable = savedObjects.level === ServiceStatusLevels.available;
+    result.next(isEsHealthy && isEsServiceAvailable && isSoServiceAvailable);
+  });
+  // Load cluster health to ensure task index is ready
+  getClusterClient()
```

Comment:
Added some logging here to see what this is doing.  After starting Kibana, I see this called just once - successfully :-)

But just once.  Shouldn't this be more reactive?  It's basically just run at startup ...